packages/graphql/tests/integration/issues/4099.int.test.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { graphql } from "graphql";
+import type { Driver, Session } from "neo4j-driver";
+import { Neo4jGraphQL } from "../../../src";
+import { cleanNodes } from "../../utils/clean-nodes";
+import { createBearerToken } from "../../utils/create-bearer-token";
+import { UniqueType } from "../../utils/graphql-types";
+import Neo4j from "../neo4j";
+
+describe("https://github.com/neo4j/graphql/issues/4099", () => {
+    let driver: Driver;
+    let neo4j: Neo4j;
+    let neoSchema: Neo4jGraphQL;
+    let session: Session;
+    const secret = "secret";
+
+    let User: UniqueType;
+
+    beforeAll(async () => {
+        neo4j = new Neo4j();
+        driver = await neo4j.getDriver();
+    });
+
+    beforeEach(async () => {
+        session = await neo4j.getSession();
+
+        User = new UniqueType("User");
+
+        const typeDefs = /* GraphQL */ `
+            type JWT @jwt {
+                isAdmin: Boolean!
+            }
+
+            type ${User} @authorization(filter: [{ operations: [READ], where: { jwt: { isAdmin: true } } }]) {
```

Comment:
Is it worth adding another test for `isAdmin_NOT: false` just since I observed the _NOT variant also having an undefined operator without any way of coalescing it like for `EQ`?

I don't know if this is somehow handled elsewhere, which is why I didn't open this PR myself

Reply:
Thank you @jbhurruth, you just caught something else there! The code was actually missing a bit of functionality for `NOT` filters, and this has been added now.